src/factories/StoneFactory.js

```diff
@@ -1,20 +1,32 @@
-import { coal, stone } from '../data/ores'
-import { COAL_SPAWN_CHANCE } from '../constants'
+/** @typedef {import("../index").farmhand.item} farmhand.item */
 import { randomNumberService } from '../common/services/randomNumber'
+import { Factory } from '../interfaces/Factory'
+import { coal, saltRock, stone } from '../data/ores'
+import {
+  COAL_SPAWN_CHANCE,
+  SALT_ROCK_SPAWN_CHANCE,
+  STONE_SPAWN_CHANCE,
+} from '../constants'
 
 /**
  * Resource factory used for spawning stone
  * @constructor
  */
-export default class StoneFactory {
+export default class StoneFactory extends Factory {
   /**
    * Generate resources
-   * @returns {Array} an array of stone and coal resources
+   * @returns {Array.<farmhand.item>} an array of stone and coal resources
    */
   generate() {
     let resources = []
 
-    resources.push(this.spawnStone())
+    if (randomNumberService.isRandomNumberLessThan(STONE_SPAWN_CHANCE)) {
+      resources.push(this.spawnStone())
+    }
+
+    if (randomNumberService.isRandomNumberLessThan(SALT_ROCK_SPAWN_CHANCE)) {
+      resources.push(this.spawnSaltRock())
+    }
```

Comment:
This conditional check is new. It's a very subtle change to Stone spawning logic that I expect most players won't even notice.